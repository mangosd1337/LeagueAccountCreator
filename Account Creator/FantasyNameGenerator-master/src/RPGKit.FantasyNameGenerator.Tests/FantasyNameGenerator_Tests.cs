using ApprovalTests.Reporters;
using NUnit.Framework;
using RPGKit.FantasyNameGenerator.Generators;

namespace RPGKit.FantasyNameGenerator.Tests
{
    public class FantasyNameGenerator_Tests
    {
        [Test]
        [UseReporter(typeof(DiffReporter))]
        public void Male_Warrior()
        {
            // arrange
            NameGenerator.GetRandomNumber = (maxValue) => 1;
            IFantasyNameGenerator fantasyNameGenerator = FantasyNameGenerator.FromSettingsInfo(new FantasyNameSettings(Classes.Warrior, Race.None, true, true, Gender.Male));

            // act
            var names = fantasyNameGenerator.GetFantasyNames(2);

            // assert
            ApprovalTests.Approvals.VerifyAll("Male Warrior Names", names, "names");
        }

        [Test]
        [UseReporter(typeof(DiffReporter))]
        public void Female_Warrior()
        {
            // arrange
            NameGenerator.GetRandomNumber = (maxValue) => 0;
            IFantasyNameGenerator fantasyNameGenerator = FantasyNameGenerator.FromSettingsInfo(new FantasyNameSettings(Classes.Warrior, Race.None, true, true, Gender.Female));

            // act
            var names = fantasyNameGenerator.GetFantasyNames(1);

            // assert
            ApprovalTests.Approvals.VerifyAll("Female Warrior Names", names, "names");
        }

        [Test]
        [UseReporter(typeof(DiffReporter))]
        public void Female_Goblin_Warrior()
        {
            // arrange
            NameGenerator.GetRandomNumber = (maxValue) => 1;
            IFantasyNameGenerator fantasyNameGenerator = FantasyNameGenerator.FromSettingsInfo(new FantasyNameSettings(Classes.Warrior, Race.Goblin, true, true, Gender.Female));

            // act
            var names = fantasyNameGenerator.GetFantasyNames(2);

            // assert
            ApprovalTests.Approvals.VerifyAll("Female Goblin Names", names, "names");
        }
    }
}
