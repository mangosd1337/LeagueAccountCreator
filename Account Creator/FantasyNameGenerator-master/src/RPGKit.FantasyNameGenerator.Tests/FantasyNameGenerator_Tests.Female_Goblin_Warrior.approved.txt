Female Goblin Names

names[0] = Female Solea Dreadbite the Crazed of Broken Hills
names[1] = Female Solea Dreadbite the Crazed of Broken Hills
